lib/ui/painting.dart

```diff
@@ -3746,6 +3746,96 @@ class ImageShader extends Shader {
   void _initWithImage(_Image image, int tmx, int tmy, int filterQualityIndex, Float64List matrix4) native 'ImageShader_initWithImage';
 }
 
+/// A shader (as used by [Paint.shader]) that runs provided SPIR-V code.
+///
+/// NOTE: This API is in beta and does not yet work on web.
+/// See https://github.com/flutter/flutter/projects/207 for roadmap.
+///
+/// When initializing or updating the [floatUniforms], the length of float
+/// uniforms must match the total number of floats defined as uniforms in
+/// the shader. They will be updated in the order that they are defined.
+///
+/// For example, if there are 3 uniforms: 1 of type float, 1 type float2/vec2,
+/// and 1 of type vec3/float3, and 1 mat2x2 then the length of [floatUniforms]
+/// must be 10.
+///
+/// The uniforms could be updated as follows:
+///
+///   Example fragment shader uniforms.
```

Comment:
indentation in markdown is unlikely to do what you want

Reply:
removed, generally improved the docs